ui/StatusQ/include/StatusQ/aggregator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+
+class Aggregator : public QObject {
+    Q_OBJECT
+
+    Q_PROPERTY(QAbstractItemModel* model READ model WRITE setModel NOTIFY modelChanged)
+    Q_PROPERTY(QString roleName READ roleName WRITE setRoleName NOTIFY roleNameChanged)
+    Q_PROPERTY(QVariant value READ value NOTIFY valueChanged)
+
+public:
+    explicit Aggregator(QObject *parent = nullptr);
+
+    QAbstractItemModel* model() const;
+    void setModel(QAbstractItemModel* model);
+
+    QString roleName() const;
+    void setRoleName(const QString &roleName);
+
+    QVariant value() const;
+
+signals:
+    void modelChanged();
+    void roleNameChanged();
+    void valueChanged();
+
+private slots:
+    virtual void calculateAggregation() = 0;
```

Comment:
Mmm.. yeah, we could pass the role names from outside, however if we keep the `m_roleName` or the `m_rolesNames list` as a member of this class, you can always access to these roles in the method implementation and skip computation.

Reply:
We cannot skip, because we don't know which signal triggered the slot. E.g. for insertion, roles doesn't matter. But for dataChanged we can skip if relevant role is not modified.

Reply:
I've just followed your idea of adding an `acceptRoles` method and:
```
connect(m_model, &QAbstractItemModel::dataChanged, this,
              [this](const QModelIndex& topLeft, const QModelIndex& bottomRight, const QVector<int>& roles) {
          if (this->acceptRoles(roles))
              this->calculateAggregation();
        });
```
Thanks for the suggestion!